tests/baselines/reference/declarationEmitLateBoundAssignments2.js

```diff
@@ -0,0 +1,186 @@
+
+export function decl4() {}
+decl4[num] = 0
+
+export function decl5() {}
+decl5["101"] = 0
+
+export function decl6() {}
+decl6[numStr] = 0
+
+export function decl7() {}
+decl7["qwe rty"] = 0
+
+export function decl8() {}
+decl8[withWhitespace] = 0
+
+export function decl9() {}
+decl9["🤪"] = 0
+
+export function decl10() {}
+decl10[emoji] = 0
+
+export const arrow = () => {}
+arrow["B"] = 'bar'
+
+export const arrow2 = () => {}
+arrow2[c] = 100
+
+export const arrow3 = () => {}
+arrow3[77] = 0
+
+export const arrow4 = () => {}
+arrow4[num] = 0
+
+export const arrow5 = () => {}
+arrow5["101"] = 0
+
+export const arrow6 = () => {}
+arrow6[numStr] = 0
+
+export const arrow7 = () => {}
+arrow7["qwe rty"] = 0
+
+export const arrow8 = () => {}
+arrow8[withWhitespace] = 0
+
+export const arrow9 = () => {}
+arrow9["🤪"] = 0
+
+export const arrow10 = () => {}
+arrow10[emoji] = 0
+
+
+//// [declarationEmitLateBoundAssignments2.js]
+// https://github.com/microsoft/TypeScript/issues/54811
+const c = "C";
+const num = 1;
+const numStr = "10";
+const withWhitespace = "foo bar";
+const emoji = "🤷‍♂️";
+export function decl() { }
+decl["B"] = 'foo';
+export function decl2() { }
+decl2[c] = 0;
+export function decl3() { }
+decl3[77] = 0;
+export function decl4() { }
+decl4[num] = 0;
+export function decl5() { }
+decl5["101"] = 0;
+export function decl6() { }
+decl6[numStr] = 0;
+export function decl7() { }
+decl7["qwe rty"] = 0;
+export function decl8() { }
+decl8[withWhitespace] = 0;
+export function decl9() { }
+decl9["🤪"] = 0;
+export function decl10() { }
+decl10[emoji] = 0;
+export const arrow = () => { };
+arrow["B"] = 'bar';
+export const arrow2 = () => { };
+arrow2[c] = 100;
+export const arrow3 = () => { };
+arrow3[77] = 0;
+export const arrow4 = () => { };
+arrow4[num] = 0;
+export const arrow5 = () => { };
+arrow5["101"] = 0;
+export const arrow6 = () => { };
+arrow6[numStr] = 0;
+export const arrow7 = () => { };
+arrow7["qwe rty"] = 0;
+export const arrow8 = () => { };
+arrow8[withWhitespace] = 0;
+export const arrow9 = () => { };
+arrow9["🤪"] = 0;
+export const arrow10 = () => { };
+arrow10[emoji] = 0;
+
+
+//// [declarationEmitLateBoundAssignments2.d.ts]
+export declare function decl(): void;
+export declare namespace decl {
+    var B: string;
+}
+export declare function decl2(): void;
+export declare namespace decl2 {
+    var C: number;
+}
+export declare function decl3(): void;
+export declare namespace decl3 { }
+export declare function decl4(): void;
+export declare namespace decl4 { }
+export declare function decl5(): void;
+export declare namespace decl5 { }
+export declare function decl6(): void;
+export declare namespace decl6 { }
+export declare function decl7(): void;
+export declare namespace decl7 { }
+export declare function decl8(): void;
+export declare namespace decl8 { }
+export declare function decl9(): void;
+export declare namespace decl9 { }
+export declare function decl10(): void;
+export declare namespace decl10 { }

```

Comment:
those don't work because function declarations are emitted as namespaces and non-identifiers can't be exported from namespaces, see: https://github.com/microsoft/TypeScript/issues/55190

I could make an effort to avoid those empty namespaces. The issue itself stands though. I think it's a slight problem that function declarations and function expressions have such a subtle and silent (!) difference in their capabilities to emit expando members